pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts

```diff
@@ -103,3 +104,19 @@ export const download = (blob: Block): string => {
 
   return url;
 };
+
+export const getBlockByUlid = (blocks: Block[], ulid: string): Block[] => {
+  if (ulid === '') {
+    return blocks;
+  }
+
+  const ulidArray = blocks.map((block) => block.ulid);
+  const fuz = new Fuzzy({ caseSensitive: true });
+
+  const result: FuzzyResult[] = fuz.filter(ulid, ulidArray);
+
+  const resultIndex = result.map((value) => value.index);
+
+  const blockResult = blocks.filter((block, index) => resultIndex.includes(index));
+  return blockResult;
```

Comment:
I spoke with @adzshaf and mentioned that I think this could produce confusing experience for the user. If I provide full-length ULID that matches a block name exactly, but fuzzy search returns more than one result, then I would be very confused and asume that all of the blocks have the same name, which should not really be happening. I think it's fair to return multiple blocks if the ULID does not match any block exactly, i.e. it is not full-length or has a typo.

We can do this in a follow up of course, since this UI already provides a ton of value.